File: packages/js-sdk/src/regex/detectors/jp/my-number.ts
// Japanese My Number detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { jpMyNumberChecksum } from '../../validators'

class JpMyNumberDetector extends RegexDetector {
  entityType = EntityType.JP_MY_NUMBER
  score = 0.85
  contextKeywords = ['my number', '\u30DE\u30A4\u30CA\u30F3\u30D0\u30FC', 'kojin bango', 'individual number']
  contextRequired = true

  // 12 digits with optional separators
  pattern = /\b\d{4}[- ]?\d{4}[- ]?\d{4}\b/g
  validator = jpMyNumberChecksum
}

registerRegion('jp', JpMyNumberDetector)

export { JpMyNumberDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/jp/my_number.py
"""Japanese My Number detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import jp_my_number_checksum


@register_region("jp")
class JpMyNumberDetector(RegexDetector):
    entity_type = EntityType.JP_MY_NUMBER
    score = 0.85
    context_keywords = [
        "my number", "\u30de\u30a4\u30ca\u30f3\u30d0\u30fc", "kojin bango", "individual number",
    ]
    context_required = True
    context_window = 50

    # 12 digits with optional separators
    pattern = re.compile(
        r"\b\d{4}[- ]?\d{4}[- ]?\d{4}\b"
    )
    validator = staticmethod(jp_my_number_checksum)
